### PR TITLE
Discover: refactor platform styles

### PR DESCRIPTION
**Pull request by oskarth**
All styles belonging to screen/component should be in styles.cljs. This commit moves platform specific styles to appropriate place and adds deprecation notice for legacy ns.

Also fixes misc weird ns imports.

Also moves colors to common ns (FUBAR).

https://github.com/status-im/status-react/issues/2094

status: ready

### CODE REVIEW

**Review by goranjovic — APPROVED** [review]

File: src/status_im/ui/screens/discover/components/views.cljs [hunk 1]
```diff
@@ -3,35 +3,35 @@
...
+   [react/text {:style      styles/title-text
                 :font       :medium}
     (i18n/label label-kw)]
    [react/touchable-highlight {:on-press action-fn}
     [react/view {}
-     [react/text {:style (st/title-action-text active?)}
+     ;; NOTE(oskarth): text-transform to uppercase not supported as RN style
+     ;; https://github.com/facebook/react-native/issues/2088
+     [react/text {:style      (styles/title-action-text active?)
+                  :uppercase? (:uppercase? styles/subtitle-text-augment)}
```
Comment (yenda) [inline src/status_im/ui/screens/discover/components/views.cljs:24]: the workaround would be better in react/text component, might be worth a tech-debt ticket

File: src/status_im/ui/screens/discover/styles.cljs [hunk 2]
```diff
@@ -29,6 +29,44 @@
...
+;; NOTE(oskarth): Hacky use of defstyle to get
+;; platform specific styles in place where they belong
+(defstyle subtitle-text-augment
+  {:ios     {:uppercase? false}
+   :android {:uppercase? true}})
+
+(defstyle tag-view
+  {:margin-horizontal 2
+   :ios     {:flex-direction   :column
+             :background-color "rgb(227, 235, 250)"
```
Comment (yenda) [inline src/status_im/ui/screens/discover/styles.cljs]: can you move the colors to common style ? It helps keeping the number of colors in check

File: src/status_im/ui/screens/discover/styles.cljs [hunk 3]
```diff
@@ -75,12 +113,18 @@
...
-(def popular-list-container
+(defstyle popular-list-container
   {:flex             1
    :background-color :white
    :padding-top      18
    :padding-left     16
-   })
+   :ios              {:border-radius 3
+                      :border-width  1
+                      :border-color  "#D7D7D7"}
```
Comment (yenda) [inline src/status_im/ui/screens/discover/styles.cljs]: here too

File: src/status_im/ui/screens/discover/styles.cljs [hunk 4]
```diff
@@ -29,6 +29,44 @@
...
+(defstyle tag-view
+  {:margin-horizontal 2
+   :ios     {:flex-direction   :column
+             :background-color "rgb(227, 235, 250)"
+             :border-radius    4
+             :border-width     1
+             :border-color     "rgba(112, 153, 230, 0.31)"
+             :padding          6}
+   :android {:flex-direction   :column
+             :background-color "#7099e619"
```
Comment (yenda) [inline src/status_im/ui/screens/discover/styles.cljs]: here too

File: src/status_im/ui/screens/discover/styles.cljs [hunk 5]
```diff
@@ -29,6 +29,44 @@
...
+  {:ios     {:uppercase? false}
+   :android {:uppercase? true}})
+
+(defstyle tag-view
+  {:margin-horizontal 2
+   :ios     {:flex-direction   :column
+             :background-color "rgb(227, 235, 250)"
+             :border-radius    4
+             :border-width     1
+             :border-color     "rgba(112, 153, 230, 0.31)"
```
Comment (yenda) [inline src/status_im/ui/screens/discover/styles.cljs]: here too

File: src/status_im/ui/screens/discover/views.cljs [hunk 6]
```diff
@@ -1,21 +1,19 @@
...
-    [status-im.components.carousel.carousel :as carousel]
-    [status-im.ui.screens.discover.components.views :as components]
-    [status-im.ui.screens.discover.all-dapps.views :as all-dapps]
-    [status-im.utils.platform :as platform]
-    [status-im.i18n :as i18n]
-    [status-im.ui.screens.discover.styles :as styles]
-    [status-im.ui.screens.contacts.styles :as contacts-st]
-    [status-im.components.list.views :as components.list]))
+  (:require [re-frame.core :as re-frame]
+            [clojure.string :as str]
```
Comment (yenda) [inline src/status_im/ui/screens/discover/views.cljs]: string

**Review by yenda — CHANGES_REQUESTED** [review]

File: src/status_im/ui/screens/discover/components/views.cljs (hunk 1 above)
Comment (oskarth) [reply src/status_im/ui/screens/discover/components/views.cljs:24]: @yenda  Can we do this in a separate issue? This is improvement over current state of affairs

File: src/status_im/ui/screens/discover/styles.cljs (hunk 2 above)
Comment (oskarth) [reply src/status_im/ui/screens/discover/styles.cljs]: done

File: src/status_im/ui/screens/discover/styles.cljs (hunk 5 above)
Comment (oskarth) [reply src/status_im/ui/screens/discover/styles.cljs]: done

File: src/status_im/ui/screens/discover/views.cljs (hunk 6 above)
Comment (oskarth) [reply src/status_im/ui/screens/discover/views.cljs]: done

File: src/status_im/components/styles.cljs [hunk 7]
```diff
@@ -26,6 +28,7 @@
 (def color-gray8 "#6E777E")
 (def color-gray9 "#E9EBEC")
 (def color-gray10 "#9BA3A8")
+(def color-gray11 "rgb(238, 241, 245)")
```
Comment (yenda) [inline src/status_im/components/styles.cljs]: could you convert the rgb to hex ? we usually don't put rgb here

Comment (oskarth) [reply src/status_im/components/styles.cljs]: done

**Review by yenda — APPROVED** [review]

**Pull request merged**